lib/plugins/aws/invokeLocal/java/src/main/java/com/serverless/InvokeBridge.java

```diff
@@ -65,6 +65,7 @@ private Context getContext(HashMap<String, Object> parsedInput) {
   private Object getInstance() throws Exception {
     URL[] urls = {this.artifact.toURI().toURL()};
     URLClassLoader child = new URLClassLoader(urls, this.getClass().getClassLoader());
+    Thread.currentThread().setContextClassLoader(child);
```

Comment:
Could you explain how this change will resolve the issue, for people without JVM programming background?

Reply:
It's not easy to explain it to people who is not familier with Java Classloader Hierarchy, but I will try.

InvokeBridge uses URLClassLoader to load a handler class as below.

https://github.com/serverless/serverless/blob/master/lib/plugins/aws/invokeLocal/java/src/main/java/com/serverless/InvokeBridge.java#L65-L72

```
  private Object getInstance() throws Exception {
    URL[] urls = {this.artifact.toURI().toURL()};
    URLClassLoader child = new URLClassLoader(urls, this.getClass().getClassLoader());

    this.clazz = Class.forName(this.className, true, child);

    return this.clazz.newInstance();
  }
```

The groovy runtime tries to load a class such as FastStringService by using ContextClassLoder, but it fails bacause the ContextClassLoader is not a same instance of URLClassLoader. Java Classloader mechanism doesn't allow it.

https://github.com/apache/groovy/blob/master/subprojects/groovy-json/src/main/java/org/apache/groovy/json/internal/FastStringUtils.java#L35-L52

```
        private static FastStringService loadService() {
// left classloading very simple in light of potential changes needed for jdk9
// that means you might need @GrabConfig(systemClassLoader=true) if getting json via grab
//            ClassLoader rootLoader = DefaultGroovyMethods.getRootLoader(loader);
            ServiceLoader<FastStringServiceFactory> serviceLoader = ServiceLoader.load(FastStringServiceFactory.class);
            FastStringService found = null;
            for (FastStringServiceFactory factory : serviceLoader) {
                FastStringService service = factory.getService();
                if (service != null) {
                    found = service;
                    if (!(service instanceof DefaultFastStringService)) {
                        break;
                    }
                }
            }
            return found;
        }
    }
```

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ServiceLoader.html#load(java.lang.Class)

```
Creates a new service loader for the given service type, using the current thread's context class loader.
An invocation of this convenience method of the form


     ServiceLoader.load(service)
 
is equivalent to

     ServiceLoader.load(service, Thread.currentThread().getContextClassLoader())
```

Does it make sense?

Reply:
I have never used Clojure, but it seems to use ContextClassLoader by default.

* https://groups.google.com/forum/#!topic/clojure/Aa04E9aJRog
* https://clojurians-log.clojureverse.org/clojure/2020-02-03
* https://stackoverflow.com/questions/12146761/reloading-clojure-code-in-tomcat

Reply:
Thanks for detail !

Since I (contributor) am not using serverless framework with JVM-language, could you test that this changes fix the issue for Groovy ?
You can install serverless from your branch by `npm install yyamano/serverless#issue-7954` 
If you want to install serverless globally, `-g` flag is required.

Reply:
Yes, I confirmed aws-groovy-gradle and aws-clojure-gradle work fine.